scene/main/window.cpp

```diff
@@ -835,6 +853,11 @@ void Window::hide() {
 
 void Window::set_visible(bool p_visible) {
 	ERR_MAIN_THREAD_GUARD;
+	if (native_surface.is_valid()) {
+		visible = true;
```

Comment:
I understand that the native surface is handled somewhere else, but there has to be a way to tell if its window is hidden or not. The `is_visible` method can be modified to check the native surface (here or in another PR) and this assignment to `true` can be dropped.